tests/components/flic/test_binary_sensor.py

```diff
@@ -23,38 +25,27 @@ def handle_events(self):
         pass
 
     def add_scan_wizard(self, wizard):
-        assert isinstance(wizard, _MockScanWizard)
         self.scan_wizard = wizard
 
     def add_connection_channel(self, channel):
-        assert channel is None
-
-
-class _MockScanWizard:
-    on_completed = None
+        self.channel = channel
 
 
 async def test_button_uid(hass):
     """Test UID assignment for Flic buttons."""
-    address_to_uid = {
-        "80:e4:da:78:6e:11": "flic_80e4da786e11",
-        "80:E4:DA:78:6E:12": "flic_80e4da786e12",  # Uppercase address should not change uid.
+    address_to_name = {
+        "80:e4:da:78:6e:11": "binary_sensor.flic_80e4da786e11",
+        "80:E4:DA:78:6E:12": "binary_sensor.flic_80e4da786e12",  # Uppercase address should not change uid.
     }
 
-    flic_client = _MockFlicClient(tuple(address_to_uid))
-
-    with mock.patch(
-        "homeassistant.components.flic.binary_sensor.FlicClient",
-        lambda _, __: flic_client,
-    ), mock.patch(
-        "homeassistant.components.flic.binary_sensor.FlicButton._create_channel",
-        lambda _: None,
-    ), mock.patch(
-        "homeassistant.components.flic.binary_sensor.ScanWizard",
-        _MockScanWizard,
-    ), assert_setup_component(
-        1, "binary_sensor"
-    ):
+    flic_client = _MockFlicClient(tuple(address_to_name))
+
+    with contextlib.ExitStack() as stack:
+        stack.enter_context(patch("pyflic.FlicClient", lambda _, __: flic_client))
+        stack.enter_context(patch("pyflic.ButtonConnectionChannel", MagicMock))
+        stack.enter_context(patch("pyflic.ScanWizard", MagicMock))
+        stack.enter_context(assert_setup_component(1, "binary_sensor"))
```

Comment:
We don't use the `assert_setup_component` anymore. It's legacy. It's enough to assert that an expected binary sensor is present in the state machine, `hass.states`.

Reply:
Done.